### PR TITLE
apierror: init error handling

### PR DESCRIPTION
**Pull request by cedric-parisi**
Initialise the service error handling
- Implement the APIError model to return more details
about the error.
- Set appropriate HTTP statuses per error type.

### CODE REVIEW

**Review by mvdan — APPROVED** [review]
LGTM, but perhaps take a look at serviceutil/errors.

Now that Go 1.13 is getting an error wrapping interface we'll probably move to that, but the idea of error wrapping is still the same. I wonder if we should use it here.

**Comment by mvdan** [comment]
conflicts :)

File: pkg/apierror/apierror.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	return s
+}
+
+// MarshalJSON is the custom json marshaling for an APIError.
+func (a APIError) MarshalJSON() ([]byte, error) {
+	err := map[string]interface{}{
+		"message": a.GetMessage(),
+	}
+
+	if a.GetDocURL() != "" {
```
Comment (cuonglm) [inline pkg/apierror/apierror.go]: Is this check necessary?

I think we should put doc_url to map regardless of its value, so we will have consistent structure

Comment (cedric-parisi) [reply pkg/apierror/apierror.go]: hm yes, users should at least get the `baseURL` of the error page

**Comment by cedric-parisi** [comment]
I've fixed the conflicts and integrated apierror in transactions

**Pull request merged**